compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala

```diff
@@ -357,6 +360,11 @@ private[compat] trait PackageShared {
     new RandomExtensions(self)
 }
 
+class ArrayExtensions(private val fact: Array.type) extends AnyVal {
+  def from[A: ClassTag](source: TraversableOnce[A]): Array[A] =
+    fact.apply(source.toSeq: _*)
```

Comment:
source match {
    case it: Iterable[A] => it.toArray[A]
    case _ => it.iterator.toArray[A]
  }